Markdown/developer_product_guide/glossary.md

```diff
@@ -2,160 +2,134 @@
 
 ## Common terms
 
-API - Application Program Interface. Technically this meaning encompasses any programmtic interface between an App and something else, however, in software API is usually used to refer predominantly to text-based web APIs like REST APIs or gRPC APIs.
+### **API** (Application Program Interface)
 
-### **REST:** (Representational State Transfer)
+An API is any programmatic interface between an application and something else. In software, however, API usually refers to a text-based web API like a REST API or gRPC API.
 
-REST is an architectural style for designing networked applications, typically for web services. It relies on stateless communication and uses standard HTTP methods such as GET, POST, PUT, and DELETE. REST APIs represent resources (data) via URLs, and clients interact with these resources using simple HTTP requests, allowing scalable and flexible communication between systems. RESTful services often return data in formats like JSON or XML.
-
-### **SOAP:** (Simple Object Access Protocol)
-
-SOAP is a protocol for exchanging structured information in web services, relying on XML for message formatting and typically using HTTP, SMTP, or other protocols for message transmission. It defines a set of rules for structuring messages, enabling complex interactions between applications, including security and transaction handling. SOAP is considered more rigid and heavier than REST, but offers more features like built-in error handling and extensibility.
-
-### **gRPC:** (gRPC Remote Procedure Call)
-
-gRPC is a high-performance, open-source RPC (Remote Procedure Call) framework developed by Google. It allows clients to invoke methods on servers as if they were local, enabling efficient communication between distributed systems. gRPC uses Protocol Buffers (protobuf) for defining data structure schemas and offers support for multiple languages. It is known for its speed and support for bi-directional streaming, making it a popular choice for microservices and high-throughput applications.
-
-### **CLI:** command-line interface
+### **Back end**
 
-A Command-Line Interface (CLI) is a text-based user interface used to interact with software or an operating system. Users type commands into a terminal or shell, which are executed by the system. CLIs offer a powerful way to control systems or applications with precise, scriptable inputs, but require knowledge of command syntax.
+The back end of a software application is the part a user does not interact with directly. For example, the back end of a car includes the engine, electrics, and other parts that the driver cannot interact with without taking the car apart.
```

Comment:
I really think that the examples used to explain both back end and front end should be software-based rather than car-based.

Reply:
@jgd10 I will leave this to you and mark the conversation as resolved. It can always be changed later if others prefer a software-based example.